src/main/java/io/percy/appium/providers/GenericProvider.java

```diff
@@ -120,4 +127,109 @@ public void setDebugUrl(String debugUrl) {
         this.debugUrl = debugUrl;
     }
 
+    public JSONObject findIgnoredRegions(ScreenshotOptions options) {
+        JSONArray ignoredElementsArray = new JSONArray();
+        ignoreRegionsByXpaths(ignoredElementsArray, options.getIgnoreRegionXpaths());
+        ignoreRegionsByIds(ignoredElementsArray, options.getIgnoreRegionAccessibilityIds());
+        ignoreRegionsByElement(ignoredElementsArray, options.getIgnoreRegionAppiumElements());
+        addCustomIgnoreRegions(ignoredElementsArray, options.getCustomIgnoreRegions());
+
+        JSONObject ignoredElementsLocations = new JSONObject();
+        ignoredElementsLocations.put("ignoreElementsData", ignoredElementsArray);
+
+        return ignoredElementsLocations;
+    }
+
+    public JSONObject ignoreElementObject(String selector, Point location, Dimension size) {
+        double scaleFactor = metadata.scaleFactor();
+        JSONObject coOrdinates = new JSONObject();
+        coOrdinates.put("top", location.getY() * scaleFactor);
+        coOrdinates.put("bottom", (location.getY() + size.getHeight()) * scaleFactor);
+        coOrdinates.put("left", location.getX() * scaleFactor);
+        coOrdinates.put("right", (location.getX() + size.getWidth()) * scaleFactor);
+
+        JSONObject jsonObject = new JSONObject();
+        jsonObject.put("selector", selector);
+        jsonObject.put("co_ordinates", coOrdinates);
+
+        return jsonObject;
+    }
+
+    public void ignoreRegionsByXpaths(JSONArray ignoredElementsArray, List<String> xpaths) {
+        for (String xpath : xpaths) {
+            try {
+                MobileElement element = (MobileElement) driver.findElementByXPath(xpath);
+
+                Point location = element.getLocation();
+                Dimension size = element.getSize();
+                String selector = String.format("xpath: %s", xpath);
+                JSONObject ignoredRegion = ignoreElementObject(selector, location, size);
+                ignoredElementsArray.put(ignoredRegion);
+            } catch (Exception e) {
+                AppPercy.log(String.format("Appium Element with xpath: %s not found. Ignoring this xpath.", xpath));
+                AppPercy.log(e.toString(), "debug");
+            }
+        }
+    }
+
+    public void ignoreRegionsByIds(JSONArray ignoredElementsArray, List<String> ids) {
+        for (String id : ids) {
+            try {
+                MobileElement element = (MobileElement) driver.findElementByAccessibilityId(id);
+                Point location = element.getLocation();
```

Comment:
we can reduce duplication for ignoreRegionsByXpaths & ignoreRegionsByIds

Reply:
lets keep it same in all sdks, it will only give slight refactor change which we will have to do in dotnet sdk as well to keep all sdks consistent as far as code is concerned

Reply:
done